clearwater-etcd/usr/bin/etcdwrapper

```diff
@@ -1,7 +1,25 @@
 #!/bin/bash
 
-# Tiny wrapper script to redirect etcd's stdout and stderr to a file and raise appropriate ENT logs
+# Wrapper script to redirect etcd's stdout and stderr to a file and raise appropriate ENT logs
+# This also uses flock(1) to ensure that only one etcd process can write the pidfile at once
+# See https://tobrunet.ch/2013/01/follow-up-bash-script-locking-with-flock/ for more on Bash file locking
+
+PIDFILE=/var/run/clearwater-etcd/clearwater-etcd.pid
+PIDFILE_LOCK=/var/run/clearwater-etcd/clearwater-etcd.pid.lock
+
+# Claim file handle 200, writing to the lock file, so we can reference it on
+# the next line
+exec 200>$PIDFILE_LOCK 
+
+# Exit unless we have an exclusive lock on file handle 200 (the lock file) -
+# i.e. unless no other etcd process is running
+flock --nonblock 200 || exit 99
```

Comment:
Yes, I'll add some output to  /var/log/clearwater-etcd/clearwater-etcd.log and retest.